build.gradle.kts

```diff
@@ -83,6 +83,9 @@ allprojects {
             )
         }
     }
+
+    // Enables running `./gradlew allDeps` to get a comprehensive list of dependencies for every subproject
+    tasks.register<DependencyReportTask>("allDeps") { }
```

Comment:
Yes, `./gradlew dependencies` only shows the dependencies of the target project (root if otherwise unspecified). I initially ran `./gradlew dependencies` to grep for where we transitively consumed the Apache Commons Compress library but couldn't find it because it only showed me the root project dependencies.

Running `./gradlew :runtime:protocol:http-client-engines:test-suite:dependencies` shows our use of Apache Commons Compress but then you have to _know_ that **:runtime:protocol:http-client-engines:test-suite** is where to look. This new task registers an effective alias of `dependencies` for every subproject in such a way that it can be invoked at the root project and yield the dependencies for root and every subproject.